src/System.Text.JsonLab.Dynamic/System/Text/Json/JsonDynamicObject.cs

```diff
@@ -20,6 +21,78 @@ private JsonDynamicObject(Dictionary<JsonProperty, JsonValue> properties)
             _properties = properties;
         }
 
+        private static Dictionary<Type, Dictionary<int, PropertyInfoLinkedList>> TypeCache = new Dictionary<Type, Dictionary<int, PropertyInfoLinkedList>>();
+
+        public static T Deserialize<T>(ReadOnlySpan<byte> utf8)
+        {
+            if (!TypeCache.TryGetValue(typeof(T), out Dictionary<int, PropertyInfoLinkedList> dictionary))
+            {
+                dictionary = GetTypeMap(typeof(T));
+                TypeCache.Add(typeof(T), dictionary);
+            }
+
+            T instance = Deserializer<T>.Deserialize(utf8);
+
+            var reader = new JsonReader(utf8, SymbolTable.InvariantUtf8);
+            while (reader.Read())
+            {
+                switch (reader.TokenType)
+                {
+                    case JsonTokenType.PropertyName:
+                        int key = GetHashCode(reader.Value);
+                        reader.Read(); // Move to the value token
+                        JsonValueType type = reader.ValueType;
+                        switch (type)
+                        {
+                            case JsonValueType.String:
+                                PropertyInfo pi = GetPropertyInfo(dictionary, key, reader.Value);
+                                pi.SetValue(instance, new Utf8String(reader.Value));    // TODO: Use Ref.Emit instead of Reflection
+                                break;
+                            case JsonValueType.Object: // TODO: could this be lazy? Could this reuse the root JsonObject (which would store non-allocating JsonDom)?
+                                throw new NotImplementedException("object support not implemented yet.");
+                            case JsonValueType.True:
+                                pi = GetPropertyInfo(dictionary, key, reader.Value);
+                                pi.SetValue(instance, true);
+                                break;
+                            case JsonValueType.False:
+                                pi = GetPropertyInfo(dictionary, key, reader.Value);
+                                pi.SetValue(instance, false);
+                                break;
+                            case JsonValueType.Null:
+                                pi = GetPropertyInfo(dictionary, key, reader.Value);
+                                pi.SetValue(instance, null);
+                                break;
+                            case JsonValueType.Number:
+                                pi = GetPropertyInfo(dictionary, key, reader.Value);
+                                if (!Utf8Parser.TryParse(reader.Value, out int result, out _))
+                                {
+                                    throw new InvalidCastException();
+                                }
+                                pi.SetValue(instance, result);
```

Comment:
BTW, will this deserialized work with structs? i.e. can I deserialize into a struct?

Reply:
Do you mean struct instead of a class? No, it doesn't work atm since the CreateDelegate is looking for a default constructor (which doesn't work for structs).

Also, atm, nested structs/etc. don't work either.

> case JsonValueType.Object:
    throw new NotImplementedException("object support not implemented yet.");

Reply:
Also, pi.SetValue doesn't work on structs either. The ref.emit based setters should fix that.